src/coreclr/jit/emitxarch.cpp

```diff
@@ -8392,8 +8392,58 @@ void emitter::emitDispIns(
     }
     else
     {
-        attr = id->idOpSize();
-        sstr = codeGen->genSizeStr(attr);
+        emitAttr sizeAttr = id->idOpSize();
+        attr              = sizeAttr;
+
+        switch (ins)
```

Comment:
Because these instructions deal with two sizes. For example, `vectractf128` deals with both `EA_32BYTE` (source) and `EA_16BYTE` (destination) and we need the printed registers to be correct.

Reply:
> If an instruction descriptor doesn't have the needed size attribute should we move the switch/case to a separate helper function to avoid duplication the logic here and below?

There isn't really any duplication, this section is extract and non-VEX `insert`, while the bottom section is VEX `insert`. Which get handled and which operand the replacement happens on differs between the two.